server/test/services/applicant/ApplicantServiceTest.java

```diff
@@ -2021,6 +2021,74 @@ public void relevantProgramsForApplicant_hiddenFromIndex() {
         .containsExactly(programDefinition.id());
   }
 
+  @Test
+  public void relevantProgramsForApplicant_disabled() {
+    // This ensures that the applicant can always see that draft
+    // applications for a given program, even if a newer version of the
+    // program is disabled from the index.
+
+    Applicant applicant = createTestApplicant();
+
+    // Create a submitted application based on the original version of a program.
+    Program originalProgramForDraftApp =
+        ProgramBuilder.newActiveProgram("program_for_draft")
+            .withBlock()
+            .withRequiredQuestion(testQuestionBank.applicantName())
+            .build();
+    Program originalProgramForSubmittedApp =
+        ProgramBuilder.newActiveProgram("program_for_application")
+            .withBlock()
+            .withRequiredQuestion(testQuestionBank.applicantName())
+            .build();
+    applicationRepository
+        .createOrUpdateDraft(applicant.id, originalProgramForDraftApp.id)
+        .toCompletableFuture()
+        .join();
+    applicationRepository
+        .submitApplication(applicant.id, originalProgramForSubmittedApp.id, Optional.empty())
+        .toCompletableFuture()
+        .join();
+
+    // Create a new program version.
```

Comment:
**suggestion:** Maybe add "that is disabled"?

Reply:
done